File: billing-domain/src/main/java/com/example/billing/domain/webhook/WebhookDeliveryId.java
package com.example.billing.domain.webhook;

import java.util.Objects;
import java.util.UUID;

public record WebhookDeliveryId(UUID value) {
    public WebhookDeliveryId { Objects.requireNonNull(value, "WebhookDeliveryId.value"); }
    public static WebhookDeliveryId newId() { return new WebhookDeliveryId(UUID.randomUUID()); }
    public static WebhookDeliveryId of(String s) { return new WebhookDeliveryId(UUID.fromString(s)); }
    @Override public String toString() { return value.toString(); }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/webhook/WebhookDeliveryId.kt
package com.example.billing.domain.webhook

import java.util.UUID

/**
 * Webhook delivery 식별자.
 *
 * Java 호환을 위해 value class 가 아닌 일반 data class. `@get:JvmName("value")` 로 기존
 * Java record accessor `value()` 보존.
 */
data class WebhookDeliveryId(@get:JvmName("value") val value: UUID) {

    override fun toString(): String = value.toString()

    companion object {
        @JvmStatic
        fun newId(): WebhookDeliveryId = WebhookDeliveryId(UUID.randomUUID())

        @JvmStatic
        fun of(s: String): WebhookDeliveryId = WebhookDeliveryId(UUID.fromString(s))
    }
}
